jax/scipy/linalg.py

```diff
@@ -300,11 +309,19 @@ def _precise_dot(A, B):
 @jit
 def _squaring(R, n_squarings):
   # squaring step to undo scaling
-  def my_body_fun(i,R):
-    return _precise_dot(R, R)
-  lower = jnp.zeros(1, dtype=n_squarings.dtype)
-  R = lax.fori_loop(lower[0], n_squarings, my_body_fun, R)
-  return R
+  def _squaring_precise(x):
+    return _precise_dot(x, x)
+
+  def _identity(x):
+    return x
+
+  def scan_loop(n_squaring, R):
+    def scan_f(c, i):
+      return lax.cond(i < n_squaring, _squaring_precise, _identity, c), None
+    res, _ = lax.scan(scan_f, R, jnp.arange(16))
+    return res
+
+  return scan_loop(n_squarings, R)
```

Comment:
nit: you could remove one layer of indirection here:
```suggestion
  def scan_f(c, i):
    return lax.cond(i < n_squaring, _squaring_precise, _identity, c), None
  res, _ = lax.scan(scan_f, R, jnp.arange(16))
  return res
```

Reply:
Done